### PR TITLE
Return filledAmount from fillOrderAsync

### PR DESCRIPTION
**Pull request by LogvinovLeon**
This PR:
* Returns filledAmunt from fillOrderAsync and updates tests

### CODE REVIEW

File: src/contract_wrappers/exchange_wrapper.ts [hunk 1]
```diff
@@ -144,10 +145,11 @@ export class ExchangeWrapper extends ContractWrapper {
      *                                  execution the tokens cannot be transferred.
      * @param   takerAddress            The user Ethereum address who would like to fill this order.
      *                                  Must be available via the supplied Web3.Provider passed to 0x.js.
+     * @return                          The amount of the order (in taker tokens baseUnits) that was filled.
```
Comment (fabioberger) [inline src/contract_wrappers/exchange_wrapper.ts]: `The amount of the order that was filled (in taker token baseUnits).`

File: test/exchange_wrapper_test.ts [hunk 2]
```diff
@@ -339,6 +339,24 @@ describe('ExchangeWrapper', () => {
                     expect(await zeroEx.token.getBalanceAsync(takerTokenAddress, takerAddress))
                         .to.be.bignumber.equal(fillableAmount.minus(partialFillAmount));
                 });
+                it('should return filled amount', async () => {
```
Comment (fabioberger) [inline test/exchange_wrapper_test.ts]: This test is more complex then this description. A simple `should return filled amount` test would be:

```
const signedOrder = await fillScenarios.createFillableSignedOrderAsync(
                        makerTokenAddress, takerTokenAddress, makerAddress, takerAddress, fillableAmount,
                    );
const partialFillAmount = new BigNumber(3);
const filledAmount = await zeroEx.exchange.fillOrderAsync(
    signedOrder, partialFillAmount, shouldCheckTransfer, takerAddress);
expect(filledAmount).to.be.bignumber.equal(partialFillAmount);
```

I would add that test and then rename this one to: `should return the partially filled amount if the fill amount specified is greater then the amount available`

**Comment by fabioberger** [comment]
Agreed with @abandeali1. Actually all the `batch` methods should return the amount filled too.

**Comment by abandeali1** [comment]
It's harder for the batch methods to return a single amount because it's not mandatory to use orders with the same `takerToken` (this is intentional and is also the reason they don't return a value on-chain). It could be nice to return a map of tokenAddress => filledAmount, but I don't think it's completely necessary.

**Comment by fabioberger** [comment]
Let's work on returning something useful for batch methods in a subsequent PR.

**Pull request merged**